### PR TITLE
Working on file creation and template loading.

### PR DESCRIPTION
**Pull request by branflake2267**
Progress report... I'm in the middle of the templates loading and name tokens class creation. 

I've got to figure out the dependencies for velocity 1.5 or maybe I can use a newer lib in IDEA...

### CODE REVIEW

File: gwtp-idea-plugin.iml [hunk 1]
```diff
@@ -13,6 +13,10 @@
     <orderEntry type="library" name="commons-io:commons-io:2.4" level="project" />
     <orderEntry type="library" name="org.apache.commons:commons-lang3:3.1" level="project" />
     <orderEntry type="library" name="org.apache.maven.shared:maven-invoker:2.1.1" level="project" />
+    <orderEntry type="library" name="ide-templates" level="project" />
+    <orderEntry type="library" name="oro-2.0.8" level="project" />
```
Comment (branflake2267) [inline gwtp-idea-plugin.iml:16]: I'll see about changing this to the maven format in a bit. :) That is not maven pom.xml but use the dependency loader from maven central instead.

File: .idea/libraries/apache_velocity_velocity_1_5.xml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<component name="libraryTable">
+  <library name="apache-velocity:velocity:1.5" type="repository">
+    <properties maven-id="apache-velocity:velocity:1.5" />
+    <CLASSES>
+      <root url="jar://$PROJECT_DIR$/lib/velocity-1.5.jar!/" />
+    </CLASSES>
+    <JAVADOC />
+    <SOURCES />
+  </library>
+</component>
```
Comment (spg) [inline .idea/libraries/apache_velocity_velocity_1_5.xml]: no new line at end of file

**Comment by spg** [comment]
LGTM

**Comment by meriouma** [comment]
Shouldn't you add .idea to the .gitignore?

File: src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java [hunk 3]
```diff
@@ -50,13 +56,14 @@ private void run() {
...
-//        } catch (Exception e) {
-//            warn("Could not create or find the name tokens file 'NameTokens.java': Error: " + e.toString());
-//            e.printStackTrace();
-//            return;
-//        }
+        try {
+            createNametokensFile();
+        } catch (Exception e) {
+            // TODO
+            //warn("Could not create or find the name tokens file 'NameTokens.java': Error: " + e.toString());
```
Comment (meriouma) [inline src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java]: still commented?

File: src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java [hunk 4]
```diff
@@ -123,4 +130,52 @@ private PsiPackage createPackage(PsiDirectory baseDir, String packageName) {
...
+
+    private PsiClass createNewNameTokensFile() throws Exception {
+        boolean processFileOnly = true;
+        NameTokenOptions nameTokenOptions = new NameTokenOptions();
+        nameTokenOptions.setPackageName(createdNameTokensPackage.getQualifiedName());
+        CreatedNameTokens createdNameToken;
+        try {
+            createdNameToken = CreateNameTokens.run(nameTokenOptions, true, processFileOnly);
+        } catch (Exception e) {
+            throw e;
```
Comment (meriouma) [inline src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java]: why do you catch the exception?

File: src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java [hunk 5]
```diff
@@ -123,4 +130,52 @@ private PsiPackage createPackage(PsiDirectory baseDir, String packageName) {
...
+        PsiClass unitNameTokens = null;
+        if (foundNameTokens != null && foundNameTokens.size() > 0) {
+            unitNameTokens = foundNameTokens.get(0);
+        } else {
+            unitNameTokens = createNewNameTokensFile();
+        }
+
+        if (unitNameTokens == null) {
+            // TODO
+            //warn("Could not create NameTokens.java");
```
Comment (meriouma) [inline src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java]: commented?

File: src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java [hunk 6]
```diff
@@ -123,4 +130,52 @@ private PsiPackage createPackage(PsiDirectory baseDir, String packageName) {
...
+    private void createNametokensFile() throws Exception {
+        if (!presenterConfigModel.isUsePlace()) {
+            return;
+        }
+
+        // look for existing name tokens first.
+        List<PsiClass> foundNameTokens = packageHierarchy.findClassName("NameTokens");
+
+        PsiClass unitNameTokens = null;
+        if (foundNameTokens != null && foundNameTokens.size() > 0) {
```
Comment (meriouma) [inline src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java:695]: `!foundNameTokens.isEmpty()`

File: src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java (hunk 3 above)
Comment (branflake2267) [inline src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java]: Remind me what I need to do. :)

File: src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java (hunk 4 above)
Comment (branflake2267) [inline src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java]: I had a different arrangement earlier on, and this stems from copying and iterating and not noticing I could remove it. :)

**Pull request merged**